### DISCUSSION TITLE
completion mode?

### DISCUSSION THREAD

**Discussion by buster-blue** [nushell/nushell#3202] [Q&A] [answered]
I tried looking everywhere for the difference between `circular` completion and `list` completion for auto-completion in Nushell. Could someone fill me in on how they work differently, and also on what that third `fuzzy` type of completion will do when its complete?

Also should an issue be made about adding documentation for the differences, or is it supposed to be obvious from name alone?

**Comment by sophiajt** [comment, marked as answer]
We could add some documentation to make to it clearer. I think the intent is that `circular` means that each time you do autocompletion in the same position, you rotate to the next possible completion, and `list` gives you a popup with a list of possible completions.

**Reply by buster-blue** [reply]
You're right, thanks! That's what it does. I was unaware of this, because I didn't know you had to restart Nushell for the changes to take effect.

It's odd that they are separate option though. In the Fish shell (a shell with a _wonderful_ UI) for example, pressing tab to complete a command where there are multiple options, will bring up the list of options as a list, **and** cycle through them with each additional tab press, highlighting the option currently selected in the list. Are there plans to make auto-completion work more like this in the future? I tried looking for an issue to answer this question myself, but I couldn't find it. Thanks!
